openhands/events/stream.py

```diff
@@ -117,22 +118,27 @@ def get_latest_event(self) -> Event:
     def get_latest_event_id(self) -> int:
         return self._cur_id - 1
 
-    def subscribe(self, id: EventStreamSubscriber, callback: Callable, append=False):
-        if id in self._subscribers:
-            if append:
-                self._subscribers[id].append(callback)
-            else:
-                raise ValueError('Subscriber already exists: ' + id)
-        else:
-            self._subscribers[id] = [callback]
+    def subscribe(
+        self, event_id: EventStreamSubscriber, callback: Callable, callback_id: str
+    ):
+        if event_id not in self._subscribers:
+            self._subscribers[event_id] = {}
 
-    def unsubscribe(self, id: EventStreamSubscriber):
-        if id not in self._subscribers:
-            logger.warning('Subscriber not found during unsubscribe: ' + id)
-        else:
-            self._subscribers[id].pop()
-            if len(self._subscribers[id]) == 0:
-                del self._subscribers[id]
+        if callback_id in self._subscribers[event_id]:
+            logger.warning(f'Subscribed ID already exists: {callback_id}')
+
+        self._subscribers[event_id][callback_id] = callback
+
+    def unsubscribe(self, event_id: EventStreamSubscriber, callback_id):
+        if event_id not in self._subscribers:
```

Comment:
Just a small note: could we please name `event_id` something else (`subscriber_id`?), since it's of type `EventStreamSubscriber`, and not an actual event id, right? My slow brain would find it easier to read. 😊

Same in `subscribe`.